sphinx/ext/autodoc/typehints.py

```diff
@@ -30,7 +36,9 @@ def record_typehints(app: Sphinx, objtype: str, name: str, obj: Any,
             sig = inspect.signature(obj, type_aliases=app.config.autodoc_type_aliases)
             for param in sig.parameters.values():
                 if param.annotation is not param.empty:
-                    annotation[param.name] = typing.stringify(param.annotation)
+                    prefix = __ANNOTATION_KIND_TO_PARAM_PREFIX.get(param.kind, '')
```

Comment:
This change will cause the `:param:` definition without prefixes (ex. `args`, `kwargs`). So this change is not good. I think it would be better to change `modify_field_list()` and `augment_descriptions_with_types()` instead (without modifying this `annotation` data).